### ISSUE TITLE
[Feature] Add ML Framework Recommendation Engine based on DiagnosticReport hardware profile

### ISSUE THREAD

**Issue by dhanasai2** [rishabh0510rishabh/EnvForage#411]
### Is your feature request related to a problem? Please describe.

Currently, EnvForge generates environment scripts when a user already **knows** which ML framework they want (e.g., `pytorch-cuda`, `tf-gpu`). However, first-time ML practitioners often don't know which framework suits their hardware. A user with a low-VRAM GPU or no GPU at all might unknowingly pick a CUDA-heavy profile that won't work for them, leading to a broken setup.

There is no guidance in the current flow that says: **"Given your hardware, here's what we recommend."**

---

###  Describe the solution you'd like

Implement a lightweight, **deterministic rule-based ML Framework Recommendation Engine** inside the `CompatibilityEngine` (or as a separate module `recommendation_engine.py`) that accepts a `DiagnosticReport` JSON and returns a ranked list of suitable ML profiles.

**Recommendation rules (pure Python, no AI/LLM — per AI_USAGE_POLICY.md):**

| Hardware Condition | Recommended Profile |
|---|---|
| No GPU detected | `cpu-only` (PyTorch CPU, TensorFlow CPU) |
| GPU VRAM < 4GB | `yolov8-nano`, `sklearn` |
| GPU VRAM 4–8GB | `pytorch-cuda`, `yolov8` |
| GPU VRAM > 8GB | `tf-gpu`, `pytorch-cuda`, `yolov8` |
| macOS + Apple Silicon | `pytorch-mps` |
| RAM < 8GB (any OS) | Show a low-resource warning, suggest `cpu-only` profiles |

**API change:** Add a new endpoint:
Body: { "diagnostic_report": <DiagnosticReport JSON> }
Response: { "recommended_profiles": [...], "warnings": [...] }
**CLI change:** After `envforge diagnose`, print a ` Recommended profiles:` section based on the report output.

---

###  Why this is a good beginner AI/ML issue

- No ML model training required — pure deterministic logic in Python
- Teaches contributors how ML hardware (VRAM, CUDA, RAM) maps to real framework choices
- Directly extends Phase 4 (AI Troubleshooting Layer) from the roadmap
- Small, self-contained module — no risk of breaking existing templates or safety filters
- Great way to learn how production ML tooling makes hardware-aware decisions

---

### Files likely to be touched

- `backend/app/core/compatibility_engine.py` (or a new `recommendation_engine.py`)
- `backend/app/api/v1/routes/` (new `/recommend` endpoint)
- `cli/` (add recommendation output to `envforge diagnose`)
- Schema additions for `RecommendationResponse` Pydantic model

---

###  Acceptance Criteria

- [x] A `recommend_profiles(diagnostic_report: DiagnosticReport) -> RecommendationResponse` function is implemented
- [x] Rules cover: no GPU, low VRAM, medium VRAM, high VRAM, Apple Silicon, low RAM
- [x] New `POST /api/v1/recommend` endpoint works and returns ranked profiles with warnings
- [x] `envforge diagnose` CLI prints recommended profiles after hardware detection
- [x] All logic is deterministic — no LLM calls (per `AI_USAGE_POLICY.md`)

---

### Labels
`feature`, `gssoc-26`, `good first issue`, `help wanted`, `AI/ML`, `backend`, `cli`

---

### GSSoC Assignment Request

> Hi @rishabh0510rishabh  
> I'm a GSSoC '26 contributor interested in working on this feature.
> 
> I've gone through the codebase, the `AI_USAGE_POLICY.md`, `COMPATIBILITY_ENGINE.md`, and the `ROADMAP.md`. This feature fits naturally into Phase 4 (AI Troubleshooting Layer) as a deterministic foundation layer before any LLM-based suggestions are added.
> 
> I'm a beginner looking to learn both open source contribution and practical ML/AI engineering. I'd love to get assigned to this issue. I can start with a draft PR within a few days.
> 
> Please assign this to me!

**Comment by madsysharma** [comment]
Hi @rishabh0510rishabh , I would like to work on this as part of GSSoC 2026.

**Comment by rishabh0510rishabh** [comment]
Assigned @dhanasai2
